source/vibe/web/i18n.d

```diff
@@ -191,27 +193,67 @@ package template GetTranslationContext(alias METHOD)
 
 
 private struct DeclString {
+	string context;
 	string key;
 	string value;
 }
 
-string keyToIdentifier(string key)
-{
-	enum hexdigits = "0123456789ABCDEF";
-	string ret;
-	size_t istart = 0;
-	foreach (i; 0 .. key.length) {
-		auto ch = key[i];
-		if (!(ch >= 'a' && ch <= 'z' || ch >= 'A' && ch <= 'Z')) {
-			if (i > istart) ret ~= key[istart .. i];
-			istart = i+1;
-			ret ~= "_"~hexdigits[ch%0xF]~hexdigits[ch/0x100];
-		}
-	}
-	if (istart < key.length) ret ~= key[istart .. $];
-	return ret;
-}
-
+//string keyToIdentifier(string key)
+//{
+//	enum hexdigits = "0123456789ABCDEF";
+//	string ret;
+//	size_t istart = 0;
+//	foreach (i; 0 .. key.length) {
+//		auto ch = key[i];
+//		if (!(ch >= 'a' && ch <= 'z' || ch >= 'A' && ch <= 'Z')) {
+//			if (i > istart) ret ~= key[istart .. i];
+//			istart = i+1;
+//			ret ~= "_"~hexdigits[ch%0xF]~hexdigits[ch/0x100];
+//		}
+//	}
+//	if (istart < key.length) ret ~= key[istart .. $];
+//	return ret;
+//}
+
+// Example po header
+/*
+ * # Translation of kstars.po into Spanish.
+ * # This file is distributed under the same license as the kdeedu package.
+ * # Pablo de Vicente <pablo@foo.com>, 2005, 2006, 2007, 2008.
+ * # Eloy Cuadra <eloy@bar.net>, 2007, 2008.
+ * msgid ""
+ * msgstr ""
+ * "Project-Id-Version: kstars\n"
+ * "Report-Msgid-Bugs-To: http://bugs.kde.org\n"
+ * "POT-Creation-Date: 2008-09-01 09:37+0200\n"
+ * "PO-Revision-Date: 2008-07-22 18:13+0200\n"
+ * "Last-Translator: Eloy Cuadra <eloy@bar.net>\n"
+ * "Language-Team: Spanish <kde-l10n-es@kde.org>\n"
+ * "MIME-Version: 1.0\n"
+ * "Content-Type: text/plain; charset=UTF-8\n"
+ * "Content-Transfer-Encoding: 8bit\n"
+ * "Plural-Forms: nplurals=2; plural=n != 1;\n"
+ */
+
+// PO format notes
+/*
+ * # - Translator comment
+ * #: - source reference
+ * #. - extracted comments
+ * #, - flags such as "c-format" to indicate what king of substitutions may be present
+ * #| msgid - previous string comment
+ * #~ - obsolete message
+ * msgctxt - disabmbiguating context, like variable scope (optional, defaults to null)
+ * msgid - key to translate from (required)
+ * msgid_plural - plural form of the msg id (optional)
+ * msgstr - value to translate to (required)
+ * msgstr[0] - indexed translation for handling the various plural forms
+ * msgstr[1] - ditto
+ * msgstr[2] - ditto and etc...
+ */
+
+// TODO: Handle msgctxt
```

Comment:
The msgcntxt should be complete, and that TODO can be removed. The only thing remaining should be additional overloads that support plurals and context for plurals.